addons/binding/org.openhab.binding.verisure/src/main/java/org/openhab/binding/verisure/handler/VerisureBridgeHandler.java

```diff
@@ -0,0 +1,255 @@
+        logger.debug("Initializing Verisure Binding");
+        VerisureBridgeConfiguration config = getConfigAs(VerisureBridgeConfiguration.class);
+
+        this.refresh = config.refresh;
+        this.pinCode = config.pin;
+
+        if (config.username == null || config.password == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "Must configure username and password");
+        } else {
+            session = new VerisureSession();
+            authstring = "j_username=" + config.username + "&j_password=" + config.password;
+            try {
+                session.initialize(authstring, pinCode);
+                startAutomaticRefresh();
+            } catch (Error e) {
+                logger.error("Failed", e);
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+            } catch (Exception e) {
+                logger.error("Failed", e);
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+            }
+        }
+
+    }
+
+    private void stopAutomaticRefresh() {
+        if (refreshJob != null && !refreshJob.isCancelled()) {
+            refreshJob.cancel(true);
+            refreshJob = null;
+        }
+    }
+
+    private void startAutomaticRefresh() {
+        if (refreshJob == null || refreshJob.isCancelled()) {
+            logger.debug("Scheduling at fixed rate");
+            refreshJob = scheduler.scheduleAtFixedRate(pollingRunnable, REFRESH_DELAY, refresh.intValue(),
+                    TimeUnit.SECONDS);
+        }
+    }
+
+    @Override
+    public void dispose() {
+        logger.debug("Handler disposed.");
+        stopAutomaticRefresh();
+        if (session != null) {
+            session.dispose();
+            session = null;
+        }
+    }
+
+    public boolean registerObjectStatusListener(DeviceStatusListener deviceStatusListener) {
+        return session.registerDeviceStatusListener(deviceStatusListener);
+    }
+
+    /**
+     *
+     */
+    private void updateAlarmState() {
+        try {
+            ChannelUID cuid = new ChannelUID(getThing().getUID(), CHANNEL_STATUS);
+            updateState(cuid, session.getAlarmStatus());
+
+            cuid = new ChannelUID(getThing().getUID(), CHANNEL_STATUS_NUMERIC);
+            updateState(cuid, session.getAlarmStatusNumeric());
+
+            cuid = new ChannelUID(getThing().getUID(), CHANNEL_CHANGERNAME);
+            updateState(cuid, session.getAlarmChangerName());
+
+            cuid = new ChannelUID(getThing().getUID(), CHANNEL_TIMESTAMP);
+            updateState(cuid, session.getAlarmTimestamp());
+
+            cuid = new ChannelUID(getThing().getUID(), CHANNEL_STATUS_LOCALIZED);
+            updateState(cuid, session.getAlarmObject() == null ? UnDefType.UNDEF
+                    : new StringType(session.getAlarmObject().getLabel()));
+        } catch (Exception e) {
```

Comment:
Please narrow the caught exception to only those types of exceptions that are actually thrown.